lib/puppet/util/windows/taskscheduler2.rb

```diff
@@ -0,0 +1,707 @@
+
+  # DONE
+  def activate(task)
+    raise Error.new(_('No current task scheduler. Schedule.Service is NULL.')) if @task_service.nil?
+    raise TypeError unless task.is_a?(String)
+
+    task_folder = @task_service.GetFolder(get_folder_path_from_task(task))
+
+    begin
+      @pITask = task_folder.GetTask(get_task_name_from_task(task))
+    rescue WIN32OLERuntimeError => e
+      @pITask = nil
+      # TODO win32ole errors are horrible.  Assume the task doesn't exist
+    end
+    @pITaskDefinition = nil
+    
+    @pITask
+  end
+
+  # DONE
+  def deactivate()
+    @pITask = nil
+    @pITaskDefinition = nil
+  end
+
+  # DONE
+  def definition()
+    if @pITaskDefinition.nil? && !@pITask.nil?
+      # Create a new editable Task Defintion based off of the currently activated task
+      @pITaskDefinition = @task_service.NewTask(0)
+      @pITaskDefinition.XmlText = @pITask.XML
+    end
+
+    @pITaskDefinition
+  end
+
+  # Delete the specified task name.
+  #
+  # DONE
+  def delete(task)
+    raise Error.new(_('No current task scheduler. Schedule.Service is NULL.')) if @task_service.nil?
+    raise TypeError unless task.is_a?(String)
+
+    task_folder = @task_service.GetFolder(get_folder_path_from_task(task))
+
+    result = -1
+    begin
+      result = task_folder.DeleteTask(get_task_name_from_task(task),0)
+    rescue WIN32OLERuntimeError => e
+      # TODO win32ole errors are horrible.  Assume the task doesn't exist so deletion is successful
+      return true
+    end
+
+    result == S_OK
+  end
+
+  # Execute the current task.
+  #
+  # DONE
+  def run(arguments = nil)
+    raise Error.new(_('No currently active task. ITask is NULL.')) if @pITask.nil?
+
+    @pITask.Run(arguments)
+  end
+
+  # Saves the current task. Tasks must be saved before they can be activated.
+  #
+  # DONE
+  def save()
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+
+    task_path = @pITask.nil? ? @task_defn_path : @pITask.Path
+
+    task_folder = @task_service.GetFolder(get_folder_path_from_task(task_path))
+
+    task_folder.RegisterTaskDefinition(get_task_name_from_task(task_path), 
+                                       definition, TASK_CREATE_OR_UPDATE, nil, nil,
+                                       definition.Principal.LogonType)
+  end
+
+  # Terminate the current task.
+  #
+  # DONE
+  def terminate
+    raise Error.new(_('No currently active task. ITask is NULL.')) if @pITask.nil?
+
+    @pITask.Stop(0)
+  end
+
+  # TODO  Need to use the password
+  def set_principal(user, password)
+    raise Error.new(_('No currently active task. ITask is NULL.')) if definition.nil?
+
+    if (user.nil? || user == "") && (password.nil? || password == "")
+      # Setup for the local system account
+      definition.Principal.UserId = 'SYSTEM'
+      definition.Principal.LogonType = TASK_LOGON_SERVICE_ACCOUNT
+      definition.Principal.RunLevel = TASK_RUNLEVEL_HIGHEST
+      return true
+    else
+      # TODO!!!
```

Comment:
Looks like some work to do here.. wonder if it uses `SCHED_E_ACCOUNT_INFORMATION_NOT_SET` the same way as `set_account_information` does in the old version?